### DISCUSSION TITLE
Could not specify ids in Golang SDK

### DISCUSSION THREAD

**Discussion by computerAItest** [milvus-io/milvus#4033] [Q&A and General discussion] [answered]
[milvus针对golang版本的V0.10.2 最新镜像,使用客户端,无法指定ids,提示报错,只能使用默认生成id,插入向量时候](url)

ShowCollections: 
 - feature_go
Insert vector failed: Entities IDs are auto-generated. All vectors of this collection must use auto-generated IDs.
ERROR: return id array is null
2020/10/20 10:44:03 http: panic serving 172.22.0.3:51716: runtime error: index out of range [0] with length 0
goroutine 37 [running]:
`func InsertVectors(fids []int, features [][]float64) error {
	ShowCollection()
	records := make([]milvus.Entity, len(features))
	for i := 0; i < len(features); i++ {
		//这里进行初始化第二维
		records[i].FloatData = make([]float32, len(features[i]))
		for j := 0; j < len(features[i]); j++ {
			records[i].FloatData[j] = float32(features[i][j])
		}
	}
	a := []int64{12}
	if len(records[0].FloatData)!=0{
		insertParam := milvus.InsertParam{CollectionName: TABLENAME, RecordArray: records, IDArray: a}
		idArray, status, err := client.Insert(&insertParam)
		if err != nil {
			log.Printf("insert err%+v", err)
			return err
		}
		if !status.Ok() {
			println("Insert vector failed: " + status.GetMessage())
		}
		if len(idArray) != int(len(features)) {
			println("ERROR: return id array is null")
		}`

**Comment by fishpenguin** [comment, marked as answer]
If you didn't provide ids when inserting vectors for the first time, then you cannot provide ids when inserting again later.
So please make sure that if you have provided ids at the first time you insert vectors.
